sdk/database/helper/connutil/sql.go

```diff
@@ -55,16 +56,32 @@ func (c *SQLConnectionProducer) Init(ctx context.Context, conf map[string]interf
 		return nil, fmt.Errorf("connection_url cannot be empty")
 	}
 
+	// Do not allow the username or password template pattern to be used as
+	// part of the user-supplied username or password
+	if c.DisableEscaping &&
+		strings.Contains(c.Username, "{{username}}") ||
+		strings.Contains(c.Username, "{{password}}") ||
+		strings.Contains(c.Password, "{{username}}") ||
+		strings.Contains(c.Password, "{{password}}") {
```

Comment:
Do these OR conditions need to be encased in parentheses? Or would maybe a helper function be better here? I wasn't sure if a helper function would be overkill for a single if statement.

Reply:
Addressed with https://github.com/hashicorp/vault/commit/945fa1eeaf847a019df047ac20a67ca3c4bd5910